### DISCUSSION TITLE
Thanks patrick and company --- AMAZING COURSE AND COMMUNITY

### DISCUSSION THREAD

**Discussion by YusufiMekias** [Cyfrin/foundry-full-course-cu#2212] [General]
I have been trying to find a good blockchain/solidity course for a long time with virtually no luck.  Almost all tutorials on youtube either are complete clickbait crypto influencer garbage that makes my blood boil or scam videos with ai voice over.  When I found this course I immediately got hooked.  It has frustrated me a lot because I have coded along all the way past the first course and into the second, but through the frustration I have achieved many eureka moments from fixing my own obvious bugs and typos. 
 I am just trying to say that this is an unbelievably good resource and definitely the best on the internet for learning truly how blockchain, defi and solidity works under the hood.  Thank you to the course creators for taking the time out of the goodness of their hearts to offer a free resource to the public to learn about this complicated field of computer science.  Also thank you to all the users that help each other in the discussions.  When i reach a level where I feel that i can offer advice to people just starting the courses I will also answer questions.  For now though I will only be asking them as I have many many questions.  Thanks again and have a blessed day.

**Comment by EngrPips** [comment]
Welcome aboard @YusufiMekias and we are looking forward to your questions, Friend.
